source/brailleDisplayDrivers/alva.py

```diff
@@ -204,35 +205,48 @@ def terminate(self):
 				# We must sleep after closing the COM port, as it takes some time for the device to disconnect.
 				time.sleep(self.timeout)
 
-	def _ser6SendMessage(self, cmd, value=""):
-		if isinstance(value, (int, bool)):
-			value = chr(value)
-		self._dev.write(b"\x1b{cmd}{value}".format(cmd=cmd, value=value,))
+	def _ser6SendMessage(self, cmd: bytes, value: bytes = b"") -> None:
+		if not isinstance(value, bytes):
+			raise TypeError("Expected param 'value' to have type 'bytes'")
+		self._dev.write(b"".join(["\x1b", cmd, value]))
 
-	def _ser6OnReceive(self, data):
-		if data != b"\x1b": # Escape
+	def _ser6OnReceive(self, data: bytes):
+		"""
+		@type data: bytes
+		"""
+		if data != b"\x1b":  # Escape
 			return
 		cmd = self._dev.read(1)
-		value = self._dev.read(ALVA_SER_CMD_LENGTHS[cmd])
-
-		if cmd == b"K": # Input
-			self._handleInput(ord(value[0]), ord(value[1]))
-		elif cmd == b"E": # Braille cell count
-			self.numCells = ord(value)
-		elif cmd == b"?": # Device ID
-			self._deviceId = ord(value)
-		elif cmd == b"r": # Raw keyboard messages enable/disable
-			self._rawKeyboardInput = bool(ord(value))
-		elif cmd == b"H": # Time
+		value: bytes = self._dev.read(ALVA_SER_CMD_LENGTHS[cmd])
+
+		if cmd == b"K":  # Input
+			self._handleInput(group=value[0], number=value[1])
+		elif cmd == b"E":  # Braille cell count
+			self.numCells = ord(value)  # this command only gets one byte
+		elif cmd == b"?":  # Device ID
+			self._deviceId = ord(value)  # this command only gets one byte
+		elif cmd == b"r":  # Raw keyboard messages enable/disable
+			self._rawKeyboardInput = bool(ord(value))  # this command only gets one byte
+		elif cmd == b"H":  # Time
 			# Handling time for serial displays does not block initialization if it fails.
 			self._handleTime(value)
 
-	def _hidOnReceive(self, data):
+	def _hidOnReceive(self, data: bytes):
+		"""
+		@type data: bytes
+		"""
 		reportID = data[0]
 		if reportID == ALVA_KEY_REPORT:

```

Comment:
This is going to fail I think. data[0] resulted in an int and you're comparing against a single byte in a bytes object.

Reply:
Yes, good catch. int to bytes comparisons are hard to catch, the linter seems to miss them, silly really.